### ISSUE TITLE
Playground, `ImageResizer />`: `width` and `maxWidth` issue - 

### ISSUE THREAD

**Issue by ScottAgirs** [facebook/lexical#3188]
- This issue comes from the playground example.
- The `<img />` component styling has not been altered
- I am happy to create a PR if y'all think the below is an actual issue and not something I'm missing/not understanding fully.
  - If I'm missing something could someone please elaborate on usage of this?

Lexical version: `v0.5.0`

## Steps To Reproduce / The current behavior

1. Add image with "Insert image"
2. resize the added image to _some_ size
3. resize the screen to something small and see the original image overflow-x

Enough space for set width:
<img width="395" alt="Screen Shot 2022-10-15 at 10 50 53 PM" src="https://user-images.githubusercontent.com/11737572/196005275-5980b727-e1e5-44da-a093-eeab484e271b.png">

Small screen (not enough space) for set width:
<img width="316" alt="Screen Shot 2022-10-15 at 10 50 37 PM" src="https://user-images.githubusercontent.com/11737572/196005257-d3cbd815-1474-4c00-88d6-6720823b7e3e.png">

## The expected behavior
Image to be resized to the desired size IF the screen space allows for it, otherwise shrink to `maxWidth: 100%`, which would result in something like this:

<img width="338" alt="Screen Shot 2022-10-15 at 10 51 22 PM" src="https://user-images.githubusercontent.com/11737572/196005299-d49dbb80-92ab-4852-8eb2-e26cbd38fbb5.png">

**Comment by acywatson** [comment]
Yea, this sounds like it could be improved. PR is welcome here.

**Comment by thegreatercurve** [comment]
I'm not sure if this is a bug or the expected UX. I think this would be the expected behaviour if the editor was in readonly mode, in which case we could make the image responsive for smaller viewports, but if we want the user to explicitly set the pixel value of the underlying image, we shouldn't probably then override that if the editor viewport becomes smaller (with I'm presuming an intersection observer?).

**Comment by ScottAgirs** [comment]
> I'm not sure if this is a bug or the expected UX. I think this would be the expected behaviour if the editor was in readonly mode, in which case we could make the image responsive for smaller viewports, but if we want the user to explicitly set the pixel value of the underlying image, we shouldn't probably then override that if the editor viewport becomes smaller (with I'm presuming an intersection observer?).

Hmm, interesting... I wonder how commonly would this kind of behaviour would be desirable? Wouldn't in most cases it would be expected for the UI to be easy to use in favour of Figma-like - for the sake of argument - infinite overflow?

**Comment by acywatson** [comment]
> I'm not sure if this is a bug or the expected UX. I think this would be the expected behaviour if the editor was in readonly mode, in which case we could make the image responsive for smaller viewports, but if we want the user to explicitly set the pixel value of the underlying image, we shouldn't probably then override that if the editor viewport becomes smaller (with I'm presuming an intersection observer?).

I don't think you need an intersection observer for this. It's just about preserving the aspect ratio of the img AFAICT, which should be something we can do with CSS.

**Comment by ScottAgirs** [comment]
Hey Acy,

Wooho, sounds great, will do!
I’m taking off for conference travels for next 2 weeks, can’t wait to be back and set up a Zoom coffee “)

Cheers,
@SᴄᴏᴛᴛAɢɪʀs
